include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h

```diff
@@ -235,3 +235,18 @@
 // Use this config to control the minimum size of the initializer when externalizing it during serialization
 static const char* const kOrtSessionOptionsOptimizedModelExternalInitializersMinSizeInBytes =
     "session.optimized_model_external_initializers_min_size_in_bytes";
+
+// Enable EP context feature to dump the partitioned graph which include the EP context into Onnx file.
+// The dumped Onnx model with EP context can be used for future inference to avoid the EP graph partitioning/compile overhead.
+// "0": disable. (default)
+// "1": enable.
+static const char* const kOrtSessionOptionEpContextEnable = "ep.context_enable";
```

Comment:
need to add CPI for these new options

Reply:
realized that AddSessionConfigEntry can do the job

---
In reply to: [1431842901](https://github.com/microsoft/onnxruntime/pull/18877#discussion_r1431842901) [](ancestors = 1431842901)

Reply:
This seems specific to QNN so a prefix of 'qnn.' instead of 'session.' will be much clearer

Reply:
It can be re-used by other EPs like TensorRT EP.

---
In reply to: [1432239944](https://github.com/microsoft/onnxruntime/pull/18877#discussion_r1432239944) [](ancestors = 1432239944)

Reply:
> It can be re-used by other EPs like TensorRT EP.
> 
> In reply to: [1432239944](https://github.com/microsoft/onnxruntime/pull/18877#discussion_r1432239944) [](ancestors = 1432239944)

it will be used by TensorRT EP. Currently it also uses EPContext, but to support more than one partition, we will need this as well.